src/classes/PS_Request.cls

```diff
@@ -0,0 +1,298 @@
+                value = '/v1/payments/verified/purchase';
+
+            } else if (endpoint == ElevateEndpoint.COMMITMENT) {
+                value = '/v1/payments/verified/commitments';
+
+            } else if (endpoint == ElevateEndpoint.COMMITMENT_CANCEL) {
+                value = '/v1/payments/verified/commitments/' + commitmentId + '/cancel';
+            }
+
+            return String.isBlank(value)
+                ? null
+                : (configService.getEndpointRoot() + value);
+        }
+
+        /***
+        * @description Constructs JSON Web Token 
+        * @param endpoint API endpoint
+        * @return String JSON Web Token
+        */
+        private String getJwt(ElevateEndpoint endpoint) {
+            JWTPayload payload = new JWTPayload(configService.getProductId(), endpoint);
+
+            String jwt = UTIL_Jwt.getSignedJWTFromString(
+                payload.getAsString(),
+                configService.getAPIKey()
+            );
+
+            return jwt;
+        }
+    }
+
+    /***
+    * @description Assists in constructing the Payload portion of a JSON Web Token.
+    */
+    public class JWTPayload {
+        String sfdoId;
+        ElevateEndpoint endpoint;
+
+        /***
+        * @description Constructor 
+        * @param sfdoId SFDO Id
+        * @param endpoint API endpoint
+        */
+        public JWTPayload(String sfdoId, ElevateEndpoint endpoint) {
+            this.sfdoId = sfdoId;
+            this.endpoint = endpoint;
+        }
+
+        /***
+        * @description Sets JSON Web Token params and values
+        * @return String JSON Web Token
+        */
+        public String getAsString() {
+            JSONGenerator generator = JSON.createGenerator(false);
+
+            generator.writeStartObject();
+            generator.writeStringField(JWT_ISSUER_PARAM, 'ApiKeyManager'); 
+            generator.writeNumberField(JWT_EXPIRES_AT_PARAM, getExpirationDate(1)); 
+            generator.writeStringField(JWT_SUBJECT_PARAM, UserInfo.getUserName()); 
+
+            if (sfdoId != null) {
+                generator.writeStringField(JWT_SFDO_ID_PARAM, sfdoId);
+            }
+
+            generator.writeStringField(JWT_PRODUCT_PARAM, ProductType.CRM.name());
+
+            setRoles(generator);
+
+            return generator.getAsString();
+        }
+
+        /***
+        * @description Calculates expiration date from today
+        * @param daysFromNow Number of days from now
+        * @return Long JSON Web Token expiration date
+        */
+        @TestVisible
+        private Long getExpirationDate(Integer daysFromNow) {
+            DateTime expirationDate = DateTime.now().addDays(daysFromNow);
+
+            return getEpochTimestamp(expirationDate);
+        }
+
+        /***
+        * @description Calculates Epoch timestamp
+        * @param dt Datetime value
+        * @return Long 
+        */
+        private Long getEpochTimestamp(DateTime dt) {
```

Comment:
This same method (and others here)  exists in `GE_PaymentServices`. Is there opportunity to consolidate some of the duplicate  methods into a new/shared class?

Reply:
The new shared class is supposed to be `PS_Request` mainly and `UITL_Http`. I did not change GE code since its change would prolong code review and testing of the Recurring Donation work. It would be the best to create new WI/PR off the `feature/RD2` once this PR is merged and update GE code. Thoughts, @force2b , @ptewson-sfdo ?

Reply:
Fair enough. Though, we really should bring the teams together to talk about merging duplicate Elevate integration code. NPSP should have a single set of logic for Elevate integration vs. different sets of logic by feature.